fastly/resource_fastly_service_v1_splunk_test.go

```diff
@@ -0,0 +1,404 @@
+package fastly
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestResourceFastlyFlattenSplunk(t *testing.T) {
+	cases := []struct {
+		remote []*gofastly.Splunk
+		local  []map[string]interface{}
+	}{
+		{
+			remote: []*gofastly.Splunk{
+				{
+					Name:              "test-splunk",
+					URL:               "https://mysplunkendpoint.example.com/services/collector/event",
+					Format:            "%h %l %u %t \"%r\" %>s %b",
+					FormatVersion:     1,
+					ResponseCondition: "error_response",
+					Placement:         "waf_debug",
+					Token:             "test-token",
+				},
+			},
+			local: []map[string]interface{}{
+				{
+					"name":               "test-splunk",
+					"url":                "https://mysplunkendpoint.example.com/services/collector/event",
+					"format":             "%h %l %u %t \"%r\" %>s %b",
+					"format_version":     uint(1),
+					"response_condition": "error_response",
+					"placement":          "waf_debug",
+					"token":              "test-token",
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		out := flattenSplunks(c.remote)
+		if !reflect.DeepEqual(out, c.local) {
+			t.Fatalf("Error matching:\nexpected: %#v\ngot: %#v", c.local, out)
+		}
+	}
+}
+
+func TestAccFastlyServiceV1_splunk_complete(t *testing.T) {
```

Comment:
For me, it is a bit hard to understand what the acceptance tests are covering.  If I read this then I understand that the servicev1 resource is being tested with the Splunk nest block, but when I read 'complete' I am not sure about that.  The other acceptance tests end with default and env.

Reply:
This particular test applies all possible fields of the API. Therefore, I didn't want to name it `basic`. One of the biggest problems, in my opinion, is that the other suites have no tests which cover all possible fields. I'll rename it to `basic`.